stan/math/rev/core/vari.hpp

```diff
@@ -190,9 +181,7 @@ template <typename T, typename = void>
 class vari_view;
 
 template <typename T>
-class vari_view<T, require_all_t<bool_constant<!is_plain_type<T>::value>,
-                                 is_eigen_dense_base<T>>>
-    final : public vari_base {
+class vari_view<T, require_not_plain_type_t<T>> final : public vari_base {
```

Comment:
Why the change of the require? Does this now also work on some non-eigen types? Which ones?

Reply:
`rowwise()` returns a [`VectorWiseOp`](https://eigen.tuxfamily.org/dox/classEigen_1_1VectorwiseOp.html) which does not inherit from `DenseBase`. If we wanted to avoid this I think `reverse()` always returns a dense expression so for these cases we could have functions instead like `rowwise_reverse()`

Reply:
Actually I would prefere functions, such as `rowwise_reverse()`. Not a mandatory change if you disagree.

Reply:
No I like it. It's only really used for `reverse()` anyway so I think it makes sense.